api/app/utils.py

```diff
@@ -0,0 +1,18 @@
+from ast import literal_eval
+
+import requests
+from fastapi import HTTPException
+
+
+def forward_http_error(resp: requests.Response, excluded_codes: list[int]) -> None:
+    try:
+        resp.raise_for_status()
+    except requests.exceptions.HTTPError as exc:
+        code = exc.response.status_code
```

Comment:
Can't we use the resp object to get this? You should be able to use resp.status_code directly I think.
See how we handled this in this other repo [tap-github - validate_response](https://github.com/MeltanoLabs/tap-github/blob/d9ddf8464731c21eed466c603883551d59e19aaf/tap_github/client.py#L167)